ArduCopter/RC_Channel_Copter.cpp

```diff
@@ -633,6 +636,9 @@ bool RC_Channel_Copter::do_aux_function(const AuxFuncTrigger &trigger)
 
 #if AP_COPTER_AHRS_AUTO_TRIM_ENABLED
         case AUX_FUNC::AHRS_AUTO_TRIM:
+            if (!copter.flightmode->allows_auto_trim()) {
+                break;
+            }
```

Comment:
This should be down in `do_aux_function_ahrs_auto_trim`, just to keep all the logic on one spot.

Reply:
done